python/rikai/mixin.py

```diff
@@ -90,6 +90,51 @@ def __matmul__(self, style: Union[dict, "rikai.viz.Style"]) -> Drawable:
         return style(self)
 
 
+class ComposableDrawable:
+    """Mixin for a drawable that can be composed with others"""
+
+    @abstractmethod
+    def xy_min(self):
+        """return a tuple of the left top axis (xmin, ymin)"""
+
+    @abstractmethod
+    def aligned(
+        self, base_drawable: ComposableDrawable, algorithm: str = "left-top"
+    ):
+        """return a new instance that is aligned with the base drawable,
+         currently it has only `left-top` align, will
+        add more if we need them"""
+
+
+class ComposedDrawable(Drawable):
+    """A drawable that includes multi sub composable drawables,
+    will align them in rendering"""
+
+    def align(self):
+        if len(self.inner_drawables) <= 1:
+            return
+        self.inner_drawables = self.inner_drawables[:1] + [
+            inner.aligned(self.inner_drawables[0])
+            for inner in self.inner_drawables
```

Comment:
it seems the first one is aligning with itself

Reply:
Yes, all others will align with the first one, the first one won't change

Reply:
why not skip the first?

Reply:
how about skipping the first one?

Reply:
how about skipping the first one?